app/controlplane/internal/policies/registry.go

```diff
@@ -0,0 +1,66 @@
+//
+// Copyright 2024 The Chainloop Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package policies
+
+import (
+	"strings"
+
+	conf "github.com/chainloop-dev/chainloop/app/controlplane/internal/conf/controlplane/config/v1"
+)
+
+// Registry manages policy providers
+type Registry struct {
+	providers map[string]*PolicyProvider
+}
+
+func NewRegistry(conf ...*conf.PolicyProvider) *Registry {
```

Comment:
should this factory also return an error and make sure that only one is set as default and error otherwise?